core/keyboard_nav/ast_node.ts

```diff
@@ -388,7 +388,20 @@ export class ASTNode {
 
     if (currentIndex < 0) return null;
 
-    const resultIndex = forward ? currentIndex + 1 : currentIndex - 1;
+    let resultIndex = forward ? currentIndex + 1 : currentIndex - 1;
+    // The flyout may contain non-navigable elements like spacers or custom
+    // items. If the next/previous element is one of those, keep looking until a
+    // block or button is encountered.
+    while (
+      resultIndex >= 0 &&
+      resultIndex < flyoutContents.length &&
+      !(
+        flyoutContents[resultIndex].element instanceof BlockSvg ||
```

Comment:
Can you file a bug to create an `isNavigable` API or similar for flyout items? This will do for now but eventually those custom items will need to be navigable (a text entry box for searching being the most obvious case).

Reply:
Yes, this is why I was asking about ASTNode in the chat :) I think we probably just want to add that method to IASTNodeLocation; should I just go ahead and do that in this PR?

Reply:
I think so; @cpcallen to confirm. This would be in the v12 beta so we'll still have time to change it.